docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md

```diff
@@ -15,67 +15,75 @@ author: "Erikre"
-  
- The default value for <xref:System.ServiceModel.Channels.ReliableSessionBindingElement.MaxPendingChannels%2A> is 4.  
-  
-## Reliable Sessions and Hosting  
- When Web hosting a service that uses reliable sessions, you should keep the following important considerations in mind:  
-  
--   Reliable sessions are stateful, and state is maintained in the AppDomain. This means that all messages that are part of a reliable session must be processed in the same AppDomain. Web farms and Web gardens where the size of the farm or garden is > 1 can't guarantee keeping with this constraint.  
-  
--   Reliable sessions using dual HTTP channels (for example, using `WsDualHttpBinding`) can require more than the default HTTP connections per-client of 2. This means a duplex reliable session can require up to 2 connections each way, because concurrent application and protocol messages may be transferring each way at any given time. This means that, under certain conditions, depending on the message exchange pattern of the service, it is possible to deadlock a Web-hosted service using dual HTTP and reliable sessions. To increase the number of allowable HTTP connections per client, add the following to the relevant configuration file (for example, web.config of the service in question):  
-  
-```xml  
-<configuration>  
-   <system.net>  
-      <connectionManagement>  
-         <add name = "*" maxconnection = "XX" />  
-      </connectionManagement>  
-   </system.net>  
-</configuration>  
-```  
-  
- Where "XX" is the number of connections needed. The minimum in this case should be 4.
\ No newline at end of file
+
+This topic discusses best practices for reliable sessions.
+
+## Setting MaxTransferWindowSize
+
+Reliable sessions in [!INCLUDE[indigo1](../../../../includes/indigo1-md.md)] use a transfer window to hold messages on the client and service. The configurable property <xref:System.ServiceModel.Channels.ReliableSessionBindingElement.MaxTransferWindowSize%2A> indicates how many messages the transfer window can hold.
+
+On the sender, this indicates how many messages the transfer window can hold while waiting for acknowledgements; on the receiver it indicates how many messages to buffer for the service.
+
+Choosing the right size impacts the efficiency of the network and the optimal capacity of the service. The following sections detail what to consider when choosing a value for this property and the impact of the value.
+
+The default transfer window size is eight messages.
+
+### Efficient use of the network
+
+In this context, the term *network* corresponds to everything between a client (sender) and a service (receiver) used as the basis of communication. This includes the transport connections and any intermediary or bridges in-between, including SOAP routers or HTTP proxies/firewalls.
```

Comment:
everything used as the basis of communication between a client (sender) and a service (receiver).
in-between --> in between